indico/modules/categories/controllers/display.py

```diff
@@ -382,22 +387,33 @@ def _process(self):
         return jsonify_data(flash=False, html=html)
 
 
-class RHShowPastEventsInCategory(RHDisplayCategoryBase):
-    """Set whether the past events in a category are automatically displayed or not"""
+class RHShowEventsInCategory(RHDisplayCategoryBase):
+    """Set whether the events in a category are automatically displayed or not"""
+    session_field = ''
 
-    def _show_past_events(self, show_past_events):
-        category_ids = session.setdefault('fetch_past_events_in', set())
-        if show_past_events:
+    def _show_events(self, show_events):
+        category_ids = session.setdefault(self.session_field, set())
+        if show_events:
             category_ids.add(self.category.id)
         else:
             category_ids.discard(self.category.id)
         session.modified = True
 
     def _process_DELETE(self):
-        self._show_past_events(False)
+        self._show_events(False)
 
     def _process_PUT(self):
-        self._show_past_events(True)
+        self._show_events(True)
+
+
+class RHShowFutureEventsInCategory(RHShowEventsInCategory):
+    """Set whether the past events in a category are automatically displayed or not"""
+    session_field = 'fetch_future_events_in'
```

Comment:
there should be an empty line after a class-level docstring

Reply:
done